kge/job/job.py

```diff
@@ -55,36 +58,115 @@ def __init__(self, config: Config, dataset: Dataset, parent_job: "Job" = None):
             for f in Job.job_created_hooks:
                 f(self)
 
-    def resume(self, checkpoint_file: str = None):
-        """Load job state from last or specified checkpoint.
+    @staticmethod
+    def create(
+        config: Config, dataset: Optional[Dataset] = None, parent_job=None, model=None
+    ):
+        from kge.job import TrainingJob, EvaluationJob, SearchJob
 
-        Restores all relevant state to resume a previous job. To run the restored job,
-        use :func:`run`.
+        if dataset is None:
+            dataset = Dataset.load(dataset)
+
+        job_type = config.get("job.type")
+        if job_type == "train":
+            return TrainingJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        elif job_type == "search":
+            return SearchJob.create(config, dataset, parent_job=parent_job)
+        elif job_type == "eval":
+            return EvaluationJob.create(
+                config, dataset, parent_job=parent_job, model=model
+            )
+        else:
+            raise ValueError("unknown job type")
 
-        Should set `resumed_from_job` to the job ID of the previous job.
+    @staticmethod
+    def find_and_create_from(
+        checkpoint: str = None,
+        config: Config = None,
+        dataset: Dataset = None,
+        parent_job=None,
+    ) -> Job:
+        """
+        Finds and loads the checkpoint to resume from given a config file in in the
+        experiment folder structure.
+        If a checkpoint is specified, it will be loaded. In this case no config file
+        needs to be specified.
+        Args:
+            checkpoint: path to checkpoint file
+            config: config in the experiment folder structure
+            dataset: dataset object
+            parent_job: parent job (e.g. search job)
+
+        Returns: Job based on checkpoint or new Job if no checkpoint found
 
         """
-        raise NotImplementedError
+        if checkpoint is None and config is None:
+            raise ValueError(
+                "Please provide either the config file located in the folder structure "
```

Comment:
Confusing: config may not be in a "folder structure". Just say: config or checkpoint requried.